tests/test_metrics.py

```diff
@@ -391,6 +391,35 @@ def test_toe_profile_metrics(self):
         pickle_converted_result = pickle.load(open('predictor_test.pkl', 'rb'))
         self.assertEqual(metrics, pickle_converted_result)
 
+    def test_toe_profile_prognostic_horizon(self):
+        from prog_algs.predictors import ToEPredictionProfile
+        profile = ToEPredictionProfile()  # Empty profile
+        for i in range(10):
+            # a will shift upward from 0-19 to 9-28
+            # b is always a-1
+            # c is always a * 2, and will therefore always have twice the spread
+            data = [{'a': j, 'b': j -1 , 'c': (j-4.5) * 2 + 4.5} for j in range(i, i+20)]
+            profile.add_prediction(
+                10-i,  # Time (reverse so data is decreasing)
+                UnweightedSamples(data)  # ToE Prediction
+            )
+
+        # Define test sample ground truth
+        GROUND_TRUTH = {'a': 9.0, 'b': 8.0, 'c': 18.0}
+        # Create rudimentary criteria equation
+        def criteria_eqn(toe : ToEPredictionProfile, ground_truth : dict):
+            result = {}
+            for key, value in ground_truth.items():
+                if abs(toe.mean[key] - value < 0.1):
+                    result[key] = True
+                else:
+                    result[key] = False
+            return result
+        from prog_algs.metrics import prognostic_horizon
+        # Prognostic horizon metric testing
+        ph_metrics = prognostic_horizon(profile, criteria_eqn, GROUND_TRUTH)
+        self.assertDictEqual(ph_metrics, {'c':9.0})
+
```

Comment:
Other tests we want:
* Where all criteria are met at different times
* Where 2 criteria are met at once
* Where at least one criteria is met at beginning of the prediction